### ISSUE TITLE
Move service discovery into a json dataset for easier modification / addition by contributors

### ISSUE THREAD

**Issue by codingo** [codingo/Reconnoitre#11]
Currently the file helper class does something along the lines of the following:

```
for serv in serv_dict: 
       ports = serv_dict[serv]
       if ("ftp" in serv):
           for port in ports:
               port = port.split("/")[0]
               f.write("[*] Found FTP service on %s:%s\n" % (ip_address, port))
               print("   [>] Found FTP service on %s:%s" % (ip_address, port))
               f.write("   [>] Use nmap scripts for further enumeration or hydra for password attack, e.g\n")
               f.write("      [=] nmap -sV -Pn -vv -p%s --script=ftp-anon,ftp-bounce,ftp-libopie,ftp-proftpd-backdoor,ftp-vsftpd-backdoor,ftp-vuln-cve2010-4221 -oN '%s/%s_ftp.nmap' -oX '%s/%s_ftp_nmap_scan_import.xml' %s\n" % (port, outputdir, ip_address, outputdir, ip_address, ip_address))
               f.write("      [=] hydra -L User List -P Pass List -f -o %s/%s_ftphydra.txt -u %s -s %s ft
```

This is a woefully inefficient way of assigning services and commands to nmap outputs bred from a quick time to market. The intention of this feature is to update this to work off of a json dataset where command line variables can be used to build strings.

For example,

```
{
  "name": "web service scans",
  "description": "[*] Found HTTP service on $ip:$port",
  "service-names": [
    "http",
    "ssl/http",
    "https"
  ],
  "discoveries": [
    {
      "description": "Use nikto & dirb / dirbuster for service enumeration, e.g",
      "commands": [
        "nikto -h $ip -p $port > $outputdir/$ip_nikto.txt",
        "dirb http://$ip:$port/ -o $outputdir/$ip_dirb.txt"
      ]
    },
    {
      "description": "Use curl and W3M (apt install w3m) to retreive web headers and find host information, e.g",
      "commands": [
        "curl -i $ip:$port",
        "w3m -dump $ip/robots.txt > $outputdir/$ip_robots.txt"
      ]
    }
  ]
}
```

Would store results relating to HTTP and HTTPS scans and could be easily expanded by future contributors without having to contribute to the core PR.

**Comment by codingo** [comment]
JSON structure updated in working branch: https://github.com/codingo/Reconnoitre/blob/Feature-Json-FIle-For-Services/reconnoitre/config.json

**Comment by cwaazywabbit** [comment]
Issued a pull request.

**Comment by codingo** [comment]
Thankyou! I'm onsite for a few days but I'll review and aim to merge this before Friday.

**Comment by codingo** [comment]
Closed with #30
